Fix #356: Make "Create a Good PR" tutorial

LittleRainC opened the pull request:
Fix #356: Make "Create a Good PR" tutorial -> [Make "Create a Good PR" tutorial](https://github.com/oppia/oppia-web-developer-docs/issues/356). 

Changed files:

* Add a "Create a Good PR" file - add the page of it
* Siderbar.md - add a link for "Create a Good PR" under "Making your first PR" - "Rules for making PRs"

![image](https://github.com/user-attachments/assets/3dbaf772-0953-4a4e-8587-8ddbbcb74cec)
![image](https://github.com/user-attachments/assets/b8064464-405c-420e-be06-9bc20d74d2bf)
![image](https://github.com/user-attachments/assets/383dd049-dddb-48a5-b36f-8b75844ee434)
![image](https://github.com/user-attachments/assets/1359da1b-4e4f-4c28-830f-93b58085eb78)
![image](https://github.com/user-attachments/assets/ec601dc8-9604-407e-956c-b3698d50d0a3)

LittleRainC committed:
Add "Create a Good PR" file

LittleRainC committed:
Update _Sidebar.md

add "Create a Good PR" under "Making your first PR"

LittleRainC committed:
Update Create a Good PR.md

Add table of content part

seanlip commented on _Sidebar.md:
```diff
@@ -32,6 +32,7 @@
       * [[Tips for analyzing the codebase|Analyzing-the-codebase]]
     * [[Learning resources for developers|Learning-Resources]]
     * [[Rules for making PRs|Rules-for-making-PRs]]
+      * [[Create a Good PR|Create a Good PR]]
```
Maybe we should reorganize this section...

- Good first issues (bolded)
- Tips for finding the right code to change
  - Finding the commit that introduced a bug
  - Tips for analyzing the codebase
- How to create a good PR (your new page, bolded)
  - Rules for making PRs
  - Coding style guide

and we move the learning resources for developers to the "getting started" section above? Just some thoughts, feel free to suggest a better ordering -- we want devs to discover this page. :)

LittleRainC replied:
adjusted as pointed above

LittleRainC replied:
![image](https://github.com/user-attachments/assets/870ed12d-b47b-4e18-ad24-8abd497a15e0)
adjusted as pointed

seanlip commented on Create a Good PR.md:
```diff
@@ -0,0 +1,113 @@
+- [Reviewers Appreciate...](#reviewers-appreciate)
+- [Need Help?](#need-help)
+
+This page is an introduction to creating a great Pull Request(PR). A great Pull Request (PR) is more than just code—it’s clean, clear, and easy for reviewers to understand. This guide shares best practices for writing high-quality PRs that are easy to review and quick to merge.
```
add space between "Pull Request" and "(PR)" in first sentence

LittleRainC replied:
adjusted as pointed above

LittleRainC replied:
adjusted -> [Create a Good PR](https://github.com/LittleRainC/oppia-web-developer-docs/blob/dev_branch2/Create%20a%20Good%20PR.md?plain=1)
![image](https://github.com/user-attachments/assets/31af8f2d-95e4-41be-9ff2-c505ba6ccafa)

seanlip commented on Create a Good PR.md:
```diff
@@ -0,0 +1,113 @@
+
+Contributing code is awesome—but making it **easy to review** is just as important. A thoughtful PR saves time for reviewers, avoids misunderstandings, and helps Oppia maintain a high standard of quality.
+
+## Before You Open Your PR
```
In this doc, do you need to refer to the more detailed "PR guide" at some point?

LittleRainC replied:
I added a link for the detailed PR guide as a essential checklist, and I think it might be okey?

LittleRainC replied:
I added a link for the detailed PR guide.
![image](https://github.com/user-attachments/assets/27143237-f84b-431e-b424-40e5cc20884c)

seanlip commented on Create a Good PR.md:
```diff
@@ -0,0 +1,113 @@
+
+## Final Checklist Before You Request Review
+
+- My PR addresses **one issue only**
```
Format as a checklist using `- [ ] ...`? (Not sure)

LittleRainC replied:
adjusted as pointed above

LittleRainC replied:
Replaced all the "-" with "- [ ]"
![image](https://github.com/user-attachments/assets/8c763255-76bf-4e5c-8e5a-b7a3a4b475f8)

seanlip commented on Create a Good PR.md:
```diff
@@ -0,0 +1,113 @@
+Don’t be afraid to ask! You can:
+
+- Tag reviewers in comments
+- Use GitHub Discussions
```
Link to the relevant github discussions category

LittleRainC replied:
link added

LittleRainC replied:
added that link
![image](https://github.com/user-attachments/assets/c41b9637-a9be-46fc-ab34-469df87e8297)

seanlip commented on Create a Good PR.md:
```diff
@@ -0,0 +1,113 @@
+
+- Tag reviewers in comments
+- Use GitHub Discussions
+- Ask in Oppia’s chat or community channels
```
How does this differ from the previous bullet point?

LittleRainC replied:
Removed "Ask in Oppia's chat or community channels"

LittleRainC replied:
This is really a good point. I removed "Ask in Oppia's chat or community channels"
![image](https://github.com/user-attachments/assets/abd79976-f610-4d19-955b-b1a77bf5c9b5)

seanlip reviewed (changes requested):
Thanks @LittleRainC! Left some notes, PTAL.

LittleRainC committed:
Update _Sidebar.md

Moved the learning resources for developers to the "getting started" section above. Adjusted the structure of makings PRs.

LittleRainC committed:
Update Create a Good PR.md

1. Added space between "Pull Request" and "(PR)" in first sentence
2. Added a refer to the more detailed "PR guide" at some point
3. Used a checklist format - [ ] 
4. Linked the relevant github discussions category
5. Removed the "Ask in Oppia’s chat or community channels".

seanlip commented:
@LittleRainC Please don't resolve review comments, it makes it hard for reviewers to tell what has and hasn't been addressed. Follow the instructions in step 5 [here](https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve) and make sure that the assignee is correct as well.

Please fix the above and ensure you are following the correct PR review process -- thanks.

seanlip commented on Create a Good PR.md:
```diff
@@ -0,0 +1,114 @@
+
+A good pull request isn’t just about working code—it’s about making collaboration easier. Clear, well-structured PRs save reviewers time, reduce confusion, and help maintain high code quality. In open source, thoughtful PRs build trust and keep the project maintainable for everyone.
+
+For the essential checklist and best practices, be sure to follow [Rules for making PRs](./rules-for-making-prs.md)—every good PR starts there. 
```
There's a syntax you can use for links to other pages in the same wiki -- please look at the other wiki pages for how to do this, so that the md files follow a consistent style.

LittleRainC replied:
Updated
![image](https://github.com/user-attachments/assets/e20c4568-ceac-41d6-98c4-1fc2df6a0484)

seanlip replied:
@LittleRainC The syntax is `[Rules for making PRs|Rules-for-making-PRs]`, right? Please check other wiki pages to make sure I got this correct.

LittleRainC replied:
I think it should be `[Rules for making PRs](https://github.com/oppia/oppia/wiki/Rules-for-making-PRs)"`. Because in [Tutorial-Learn-How-to-Fix-a-CI-Flake-waitFor-Statements.md](https://github.com/oppia/oppia-web-developer-docs/blob/develop/Tutorial-Learn-How-to-Fix-a-CI-Flake-waitFor-Statements.md?plain=1), you can notice that they used the same syntax.
![image](https://github.com/user-attachments/assets/2a6b1bb4-fe84-4135-967b-2a997178ecb5)

seanlip reviewed (changes requested):
Thanks, looks good! Just one small note.

seanlip commented:
@LittleRainC Are you still working on this PR?

LittleRainC committed:
Merge branch 'oppia:develop' into dev_branch2

LittleRainC committed:
Update Create a Good PR.md

Updated the link for "Rule for making PRs"

LittleRainC commented:
@seanlip Yeah, I'm just a bit busy last week. Sorry!

seanlip commented:
Thanks @LittleRainC! Please resolve the merge conflicts, and then I think we can merge this.

LittleRainC committed:
Merge branch 'develop' into dev_branch2

seanlip commented:
@LittleRainC Sorry, one more comment -- https://github.com/oppia/oppia-web-developer-docs/pull/457#discussion_r2121111617

LittleRainC committed:
Merge branch 'oppia:develop' into dev_branch2

LittleRainC commented:
@seanlip Hi, I looked up the format again, and I don't think that is a problem. What's your opinion about it?

LittleRainC commented:
@seanlip

seanlip commented:
Sorry @LittleRainC, I missed this, primarily because I wasn't listed as an "Assignee" (I only look at github.com/pulls/assigned to see what I need to review, typically).

I think what you have should be OK -- there's a different syntax that's often used for links to a page within the same wiki but it's not a big deal either way. I'll approve and we can get this merged, sorry it took so long!
